src/mig/testutil/modtest.go

```diff
@@ -0,0 +1,60 @@
+package testutil
+
+import (
+    "testing"
+    "mig"
+    "encoding/json"
+)
+
+func unexpectedErrors(t *testing.T, errors []string) {
+    for _, msg := range errors {
+        t.Errorf("Unexpected error: %s", msg)
+    }
+}
+
+func RunModule(t *testing.T, m mig.Moduler, args interface{}) (mig.ModuleResult, []byte) {
+    var marshalled_args []byte
+    marshalled_args, err := json.Marshal(args)
+    if (err != nil) {
+        t.Error(err)
+    }
+
+    t.Logf("Running module with args %s", marshalled_args)
+    marshalled_res := []byte(m.Run(marshalled_args))
```

Comment:
With the new `Run()` interface, you can do something similar but using a `bytes.Buffer` to pass the `marshalled_args`.
```go
buf := bytes.NewBuffer(marshalled_args)
marshalled_res := []byte(m.Run(buf))
```